GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -62,22 +65,28 @@ subroutine gx_minimax_grid(num_points, e_min, e_max, &
          intent(out)                                  :: omega_points, omega_weights
     real(kind=dp), allocatable, dimension(:, :), &
          intent(out)                                  :: cosft_wt, cosft_tw, sinft_wt
-    real(kind=dp), intent(out)                        :: max_errors(3), cosft_duality_error
+    real(kind=dp), allocatable, dimension(:, :), &
+         optional,intent(out)                         :: sinft_tw
+    real(kind=dp), intent(out)                        :: max_errors(4), cosft_duality_error
```

Comment:
We have to keep `max_errors` with dimension 3, it's generating error in the already implemented interfaces. You can implement the `max_error` for the sine $\omega$ to $\tau$ as an optional parameter